src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs

```diff
@@ -178,9 +178,9 @@ private async Task CopyToAsyncCore<TStream>(TStream destination, Func<TStream, R
         {
             while (true)
             {
-                SequencePosition consumed = default;
-
                 ReadResult result = await ReadAsync(cancellationToken).ConfigureAwait(false);
+                SequencePosition consumed = result.Buffer.Start;
```

Comment:
You can avoid some superfluous reads here.

You have below:

```c#
ReadOnlySequence<byte> buffer = result.Buffer;
SequencePosition position = buffer.Start;
```

Move this outside of try/catch and put below it:
```c#
SequencePosition consumed = position;
```

Reply:
Also you have below that:

```c#
if (consumed.Equals(default))
{
    consumed = buffer.End;
}
```

which is no longer necessary.

Reply:
I think the following is still necessary.

```c#
if (consumed.Equals(default))
{
    consumed = buffer.End;
}
```

ROS.TryGet sets `position` to default at the end of the sequence the last time it returns true. [ROS.TryGet only returns false after receiving an already-default `position` as an argument](https://github.com/dotnet/runtime/blob/bc47321f0c1922408fa07e1953e5b9b5735cf7d2/src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.Helpers.cs#L17-L24).

I guess we could just change it to just `consumed = buffer.End;` without the `if (consumed.Equals(default))` check since consumed should always be default after the while loop completes successfully.